pkg/apply/desiredset.go

```diff
@@ -56,6 +56,17 @@ func (o desiredSet) ApplyObjects(objs ...runtime.Object) error {
 	return o.Apply(os)
 }
 
+// WithGVK uses a known listing of existing gvks to modify the the prune types to allow for deletion of objects
+func (o desiredSet) WithGVK(gvks []schema.GroupVersionKind) Apply {
+	if o.pruneTypes == nil {
```

Comment:
WithCacheTypes will blank out this map, so you need to edit that method too.  Also this method provides a clone of the object so if pruneType != nil you need to copy the map to a new map and then set the new gvks.